doc/alertmanager-mixin/dashboards/overview.libsonnet

```diff
@@ -0,0 +1,310 @@
+local grafana = import 'github.com/grafana/grafonnet-lib/grafonnet/grafana.libsonnet';
+local dashboard = grafana.dashboard;
+local row = grafana.row;
+local prometheus = grafana.prometheus;
+local template = grafana.template;
+local graphPanel = grafana.graphPanel;
+
+
+{
+  grafanaDashboards+:: {
+    local clusterTemplate =
+      template.new(
+        name='cluster',
+        datasource='$datasource',
+        query='label_values(alertmanager_alerts, %s)' % $._config.clusterLabel,
```

Comment:
I'm not quite sure if this is doing what you want.

An _Alertmanager cluster_ is a different thing from a _Kubernetes cluster_ (or generally a cluster of nodes you run services on). In fact, for HA, an _Alertmanager cluster_ will commonly span multiple _K8s clusters_. Ideally, you have only one (global) Alertmanager cluster in your org, but should you have multiple (e.g. you have a dev cluster, or you have strictly separate network partitions, each of which needs its own Alertmanager cluster), then you want your dashboard to switch between those, and when viewing at a particular Alertmanager cluster, you want to see _all_ instances included there, even if they run in different K8s clusters.

In fact, the mixin already has this concept of an Alertmanager cluster, see `alertmanagerClusterLabels` and `alertmanagerClusterName` in `config.libsonnet`. So ideally, your multi-cluster support utilizes those. The mixin allows for multiple labels to define the Alertmanager cluster, which makes the templating here hard. But I think it's possible.

Reply:
Particularly, my setup consists of multiple k8s clusters with one alertmanager per cluster that do not communicate between themselves. I understand that my setup is not the usual Alertmanager-HA that we all should aim for, but I think I won't be able to test the filter using `alertmanagerClusterLabels` 😬

Reply:
In your case, you have "one-instance clusters", and the `alertmanagerClusterLabels` could indeed be just `cluster`.

Maybe the only problem is that the mixin as-is allows multiple labels for `alertmanagerClusterLabels` (which I introduced because it was easy at the time, and I think it's needed in some use cases). In fact, here at GL, we use `job, namespace` as the `alertmanagerClusterLabels`. `job` is usually something like `global-alertmanager` while `namespace` can be used to, for example, have a production global AM cluster and a separate test cluster.

I think this should all work, just that you need to jump through some jsonnet hoops to iterate through all the labels in `alertmanagerClusterLabels` and dynamically create the corresponding template variables for Grafana. (This will be easier if `alertmanagerClusterLabels` is a list. But we can totally make it one.)

The Grafana `query_result(query)` might also help here. See https://grafana.com/docs/grafana/latest/datasources/prometheus/#query-variable